extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -714,19 +732,21 @@ export class AmpA4A extends AMP.BaseElement {
         layoutAdPromiseDelay: Math.round(delta),
         isAmpCreative: !!creativeMetaData,
       });
-      if (creativeMetaData) {
-        // Must be an AMP creative.
-        return this.renderAmpCreative_(creativeMetaData).catch(err => {
-          // Failed to render via AMP creative path so fallback to non-AMP
-          // rendering within cross domain iframe.
-          user().error(TAG, this.element.getAttribute('type'),
-            'Error injecting creative in friendly frame', err);
-          rethrowAsync(this.promiseErrorHandler_(err));
-          return this.renderNonAmpCreative_();
-        });
+      if (!creativeMetaData) {
+        // Non-AMP creative case, will verify ad url existence.
+        return this.renderNonAmpCreative_();
+      } else if (creativeMetaData.collapse) {
```

Comment:
instead of store the boolean in the `creativeMetaData`, have you thought about storing it as a member field in the class. as you can take it as a state of this slot.